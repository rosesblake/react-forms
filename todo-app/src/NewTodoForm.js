import React, { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = {
    task: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData });
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo">New Todo:</label>
      <input
        id="todo"
        type="text"
        name="task"
        placeholder="new todo"
        value={formData.task}
        onChange={handleChange}
      />
      <button>Add Todo</button>
    </form>
  );
};

export default NewTodoForm;
