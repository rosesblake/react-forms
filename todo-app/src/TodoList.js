import React, { useState } from "react";
import Todo from "./Todo";
import { v4 as uuid } from "uuid";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  const [todos, setTodos] = useState([]);
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
  };

  return (
    <div>
      <h2>Todo List</h2>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map(({ id, task }) => (
          <li key={id}>
            <Todo task={task} id={id} />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
