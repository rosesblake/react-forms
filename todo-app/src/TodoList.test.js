import { render, fireEvent, screen } from "@testing-library/react";
import TodoList from "./TodoList";
import NewTodoForm from "./NewTodoForm";

it("renders", () => {
  render(<TodoList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("submits form with data", () => {
  render(<TodoList />);

  const input = screen.getByLabelText("New Todo:");
  const button = screen.getByRole("button");

  fireEvent.change(input, { target: { value: "buy milk" } });
  fireEvent.click(button);

  const todo = screen.getByText("buy milk");
  expect(todo).toBeInTheDocument();
});
