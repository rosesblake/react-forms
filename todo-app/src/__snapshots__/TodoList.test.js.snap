// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="todo"
      >
        New Todo:
      </label>
      <input
        id="todo"
        name="task"
        placeholder="new todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
